rock_paper_scissors/encrypted-ixs/src/lib.rs

```diff
@@ -6,30 +6,62 @@ mod circuits {
 
     // Consider 0 - Rock, 1 - Paper, 2 - Scissors
     pub struct GameMoves {
+        player_a_move: u8,
+        player_b_move: u8,
+    }
+
+    #[instruction]
+    pub fn init_game(mxe: Mxe) -> Enc<Mxe, GameMoves> {
+        let game_moves = GameMoves {
+            player_a_move: 3, // Moves are 0-2, so 3 is invalid
+            player_b_move: 3, // Moves are 0-2, so 3 is invalid
+        };
+
+        mxe.from_arcis(game_moves)
+    }
+
+    pub struct PlayerMove {
+        player: u8,
         player_move: u8,
-        house_move: u8,
     }
 
     #[instruction]
-    pub fn compare_moves(game_ctxt: Enc<Client, GameMoves>) -> u8 {
+    pub fn player_move(
+        player_move_ctxt: Enc<Client, PlayerMove>,
+        game_ctxt: Enc<Mxe, GameMoves>,
+    ) -> Enc<Mxe, GameMoves> {
+        let player_move = player_move_ctxt.to_arcis();
+        let mut game_moves = game_ctxt.to_arcis();
+
+        if player_move.player == 0 && game_moves.player_a_move == 3 {
+            game_moves.player_a_move = player_move.player_move;
+        } else if player_move.player == 1 && game_moves.player_b_move == 3 {
+            game_moves.player_b_move = player_move.player_move;
+        }
+
+        game_ctxt.owner.from_arcis(game_moves)
+    }
+
+    #[instruction]
+    pub fn compare_moves(game_ctxt: Enc<Mxe, GameMoves>) -> u8 {
         let game_moves = game_ctxt.to_arcis();
 
-        // 0 - tie, 1 - player wins, 2 - house wins
+        // 0 - tie, 1 - player A wins, 2 - player B wins
```

Comment:
maybe we should init it as 3 or something to show someone played an invalid move, otherwise if one guy played fair and the other submitted some wrong move it says tie